### ISSUE TITLE
Only part of the execute time records were shown after reopen notebook if there were more than about 20 code cells     

### ISSUE THREAD

**Issue by DatumWorld** [deshaw/jupyterlab-execute-time#61]
Reproduce:
1 create a notebook with  more than 20 code cells in it and run it,  all the execute time records for all code cells will be shown .
![image](https://user-images.githubusercontent.com/24842100/149700851-9be63ff1-e1b4-4b64-9d2f-fdf5c5335a57.png)
2  Then save and reopen the notebook, scroll down, there were  several code cells  with no execute time records shown in the tail part of the notebook 
![image](https://user-images.githubusercontent.com/24842100/149700945-b3e72353-c949-413b-b130-85300e96f879.png)
3 It occurs in both Python notebook and  R notebook.

Envs:
 win 10/2019, a fresh installation with the following code:
conda create -n jupyterpy38demo python=3.8
pip install jupyterlab==3.2.8  jupyterlab-execute-time pandas 
conda install -c r r-essentials (R 3.6.1)

Expectation:
Like it behaves in Jupyterlab 2.2.9, all the  execute time records are shown normally.

**Comment by mlucool** [comment]
Humm. This seems like it may be a jupyterlab bug. 20 sounds a bit too much like the cutoff for numberCellsToRenderDirectly. If you set that to 25 does it start at 25?

Setting: https://github.com/jupyterlab/jupyterlab/blob/5962faa9aa1626a088301b87c399df94192e98c0/packages/notebook-extension/schema/tracker.json#L855

CC @echarles

**Comment by echarles** [comment]
Let's wait on the confirmation, but indeed 20 directly drives to the virtual notebook feature. I guess the extensions listens to some signals, and maybe the lazily rendered cells (the one with index above the defined threshold numberCellsToRenderDirectly do not generate that signal).

**Comment by DatumWorld** [comment]
> Humm. This seems like it may be a jupyterlab bug. 20 sounds a bit too much like the cutoff for numberCellsToRenderDirectly. If you set that to 25 does it start at 25?
> 
> Setting: https://github.com/jupyterlab/jupyterlab/blob/5962faa9aa1626a088301b87c399df94192e98c0/packages/notebook-extension/schema/tracker.json#L855
> 
> CC @echarles

Thanks! 
I tested and found that the numberCellsToRenderDirectly value in Notebook Settings affect how many pieces of execute time  records were shown after opening one notebook .  If it was set to 30, all records were shown. While if it was set to 10,only top 10 records were shown.  The results were the same in both Python notebook and R notebook.
![image](https://user-images.githubusercontent.com/24842100/150449928-8bf20d0b-1c66-4b61-a076-9322570de551.png)
I can not decide if it is bug for  jupyterlab-execute-time or for jupyterlab ,because I know little about how they run in detail.
So,if you think it is a bug for jupyterlab ,I will open an issue there.
@mlucool @echarles

**Comment by DatumWorld** [comment]
> 

It still exists in JupyterLab 3.3.2. I set the value of numberCellsToRenderDirectly to 50 in settings now, for most of my notebooks have less than 50 cells.

**Comment by DatumWorld** [comment]
> > Humm. This seems like it may be a jupyterlab bug. 20 sounds a bit too much like the cutoff for numberCellsToRenderDirectly. If you set that to 25 does it start at 25?
> > Setting: https://github.com/jupyterlab/jupyterlab/blob/5962faa9aa1626a088301b87c399df94192e98c0/packages/notebook-extension/schema/tracker.json#L855
> > CC @echarles
> 
> Thanks! I tested and found that the numberCellsToRenderDirectly value in Notebook Settings affect how many pieces of execute time records were shown after opening one notebook . If it was set to 30, all records were shown. While if it was set to 10,only top 10 records were shown. The results were the same in both Python notebook and R notebook. ![image](https://user-images.githubusercontent.com/24842100/150449928-8bf20d0b-1c66-4b61-a076-9322570de551.png) I can not decide if it is bug for jupyterlab-execute-time or for jupyterlab ,because I know little about how they run in detail. So,if you think it is a bug for jupyterlab ,I will open an issue there. @mlucool @echarles

It still exists in JupyterLab 3.3.2. I set the value of numberCellsToRenderDirectly to 50 in settings now, for most of my notebooks have less than 50 cells.

**Comment by mlucool** [comment]
As this is a Jupyter problem, I'm closing in favor of the linked issue.
